configuration/components/loki.libsonnet

```diff
@@ -285,34 +277,19 @@ function(params) {
           // Defaults: https://github.com/grafana/loki/blob/main/production/ksonnet/loki/config.libsonnet#L181-L184
           frontend+: {
             // Mixins don't support prefixes in resources names so we have to overwrite
-            // local schedulerService = newService('query_scheduler'),
-            // scheduler_address: '%s.%s.svc.cluster.local:9095' % [schedulerService.metadata.name, loki.config.namespace],
-            // TODO(JoaoBraveCoding): replace when working on services
-            scheduler_address: '%s.%s.svc.cluster.local:9095' % [
-              loki.config.name + '-query-scheduler-grpc',
-              loki.config.namespace,
-            ],
+            local schedulerService = newService('query_scheduler'),
+            scheduler_address: '%s.%s.svc.cluster.local:9095' % [schedulerService.metadata.name, loki.config.namespace],
```

Comment:
Doesn't the scheduler service need to be headless for this configuration to work? See:

https://github.com/grafana/loki/blob/de2b0c6e9dcce028ed69c6e40f75e1ff0c081358/production/ksonnet/loki/query-scheduler.libsonnet#L7-L11

Reply:
IIUC, if we were NOT using the membership ring yes this would need to be a headless service. However, since we always run with the membership ring AFAIK this doesn't need to be a headless service. This reply is based on the fact that we got a green CI and what I understood from this comment https://github.com/observatorium/observatorium/pull/508#issuecomment-1474998694 but maybe I misunderstood it, let me know!

The mixins provides the two services https://github.com/grafana/loki/blob/de2b0c6e9dcce028ed69c6e40f75e1ff0c081358/production/ksonnet/loki/query-scheduler.libsonnet#L70-L89

Reply:
According to the docs the `frontend` block to discover scheduler pods is not part of memberlist, it simplified by using the DNS SRV records and leave anything else to GRPC: https://grafana.com/docs/loki/v2.8.x/configuration/#frontend

```yaml
# DNS hostname used for finding query-schedulers.
# CLI flag: -frontend.scheduler-address
[scheduler_address: <string> | default = ""]
```